lib/cli/interactive-setup/aws-credentials.js

```diff
@@ -86,33 +208,238 @@ const steps = {
+      let inquirerPrompt;
+
+      const timeoutDuration = 1000 * 15; // 15 seconds
+      showSkipPromptTimeout = setTimeout(() => {
+        inquirerPrompt = inquirer.prompt({
+          message:
+            '\n [If you approached an issue when setting up a provider, you may press Enter to skip this step]',
+          name: 'skipProviderSetup',
+        });
+
+        inquirerPrompt.then(() => resolve(null));
+      }, timeoutDuration);
+
+      onEvent = (event) => {
+        if (inquirerPrompt) {
+          // Disable inquirer prompt asking to skip without setting provider
+          inquirerPrompt.ui.close();
+        }
+
+        clearTimeout(showSkipPromptTimeout);
+        resolve(event);
+      };
+    });
+
+    // Listen for `provider.created` event to detect creation of new provider
+    const sdk = await getSdkInstance(orgName);
+    try {
+      await sdk.connect({
+        orgName,
+        onEvent,
+        filter: {
+          events: ['provider.created'],
+        },
+      });
+    } catch (err) {
+      // Ensure that prompt timeout is cleared in case of error
+      clearTimeout(showSkipPromptTimeout);
+
+      if (err.statusCode && err.statusCode >= 500) {
+        throw new ServerlessError(
+          'Dashboard service is currently unavailable, please try again later',
+          'DASHBOARD_UNAVAILABLE'
+        );
+      }
+      throw err;
+    }
+
+    let maybeEvent;
+    try {
+      maybeEvent = await p;
+    } finally {
+      sdk.disconnect();
+    }
+
+    if (maybeEvent) {
+      process.stdout.write(
+        `\n${chalk.green('AWS Access Role provider was successfully created')}\n`
+      );
+      return maybeEvent.data.object.provider_uid;
+    }
+
+    process.stdout.write(
+      '\nSkipping credentials provider setup. You can still setup credentials provider later.\n'
+    );
+    return null;
+  },
+  linkProviderToService: async (app, service, orgName, providerUid) => {
+    const sdk = await getSdkInstance(orgName);
+    const linkType = 'service';
+    const linkUid = getProviderLinkUid(app, service);
+    let orgUid;
+    try {
+      orgUid = await getOrgUidByName(orgName);
+    } catch (err) {
+      if (err.code === 'DASHBOARD_UNAVAILABLE') {
+        process.stdout.write(`\n${chalk.yellow(err.message)}\n`);
+        return false;
+      }
+      throw err;
+    }
+    try {
+      await sdk.createProviderLink(orgUid, linkType, linkUid, providerUid);
+      return true;
+    } catch (err) {
+      if (err.statusCode && err.statusCode >= 500) {
+        process.stdout.write(
+          `\n${chalk.yellow(
+            'Dashboard service is currently unavailable, please try again later'
+          )}\n`
+        );
+        return false;
+      }
+      throw err;
+    }
+  },
+};
+
+const isUserLoggedIn = (orgName) => {
+  if (process.env.SERVERLESS_ACCESS_KEY) {
+    return true;
+  }
+
+  const user = configUtils.getLoggedInUser();
+
+  if (user && user.accessKeys && user.accessKeys[orgName]) {
```

Comment:
Why do we need to this check? 

Should we assume that `getLoggedInUser` returns only really logged in users? Maybe it's `getLoggedInUser` that should be updated with this additional check (?)

Reply:
I think I went a bit too deep with this check as we don't strictly need the `accessKey` to be available for given org from config, as if it's missing, it will be created in the `getPlatformClientWithAccessKey` handler. I'm going to just check for existence of `user` - the logic in `getLoggedInUser` checks if user is actually logged in